code/datums/effects/water.dm

```diff
@@ -0,0 +1,27 @@
+/datum/effects/water
+	effect_name = "water"
+	duration = 20
+	icon_path = 'icons/mob/humans/onmob/water.dmi'
+	obj_icon_state_path = "+water"
+	mob_icon_state_path = "water"
+	flags = INF_DURATION
+
+/datum/effects/water/New(var/atom/A)
+	..(A)
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		H.update_effects()
+
+/datum/effects/water/Destroy()
+	if(affected_atom)
+		LAZYREMOVE(affected_atom.effects_list, src)
```

Comment:
this is already done by parent

Reply:
parent nulls `affected_atom` so I cannot call it before calling `update_effects()` to remove the spirte, nor can I leave it up to the parent's `Destroy()` since it doesnt call `update_effects()` 😔